crypto/rsa/rsa_ossl.c

```diff
@@ -482,6 +482,8 @@ static int rsa_ossl_private_decrypt(int flen, const unsigned char *from,
         r = RSA_padding_check_SSLv23(to, num, buf, j, num);
         break;
     case RSA_NO_PADDING:
+        if (j <= 0)
```

Comment:
Hmm, this by-passes my err_clear_last_constant_time
could you try this
```C
if (j > 0)
   memcpy(to, buf, (r=j));
```

Reply:
Isn't this an error case anyway, where BN_bn2binpad() failed? Maybe we should check that just after the call to BN_bn2binpad() instead?

Reply:
the error return of BN_bn2binpad is -1,
the only way I see how 0 can be returned is when num = 0, that happens
only when n=0, so that should not happen. but if it happens it would be a fatal error

Reply:
I think the other RSA_padding_check_xx test for num <= 0 and fail immediately.
not in constant time of course.

Reply:
Therefore I would not mind to handle j <= 0 in the same way.
It looks bogus to handle j == 0 and j < 0 differently.

Reply:
This really looks like one of the false positives of coverity, but I can understand that coverity can't actually see that BN_num_bytes(rsa->n) >= BN_num_bytes(ret).

Reply:
Yes, therefore I would like to not add too much branches that can't be tested by definition.

Reply:
I've removed the changes for 1453638 and 1453629 from this PR.

Somebody who better understands the implications can fix them :)
Or tag them as false positives (which also seems reasonable based on the discussion above).

Reply:
I would not have a problem when j <= 0 would be
handled exactly the same as in the pk1 ad oaep padding (it is an error),
I just was asking why j == 0 is different than j < 0.

Reply:
I've switched to checking the return of `BN_bn2binpad` and erroring for ⩽ 0.